test/distributed/_tools/test_fsdp2_mem_tracker.py

```diff
@@ -0,0 +1,238 @@
+# Owner(s): ["module: unknown"]
+import functools
+from typing import Union
+
+import torch
+import torch.nn as nn
+from torch.distributed._composable import checkpoint
+from torch.distributed._composable.fsdp import (
+    CPUOffloadPolicy,
+    fully_shard,
+    MixedPrecisionPolicy,
+    OffloadPolicy,
+)
+from torch.distributed._tensor import init_device_mesh
+from torch.distributed._tools.fsdp2_mem_tracker import FSDPMemTracker
+from torch.distributed.algorithms._checkpoint.checkpoint_wrapper import (
+    apply_activation_checkpointing,
+    CheckpointWrapper,
+)
+
+from torch.testing._internal.common_distributed import skip_if_lt_x_gpu
+from torch.testing._internal.common_fsdp import FSDPTest, MLP
+from torch.testing._internal.common_utils import run_tests
+from torch.testing._internal.distributed._tensor.common_dtensor import (
+    ModelArgs,
+    Transformer,
+    TransformerBlock,
+)
+
+c10d_ops = torch.ops.c10d
+funcol = torch.ops.c10d_functional
+
+
+class TestTrackerFullyShard1DTrainingCore(FSDPTest):
+    @property
+    def world_size(self) -> int:
+        return min(4, torch.cuda.device_count())
+
+    @skip_if_lt_x_gpu(2)
+    def test_tracker_multi_group_eager(self):
+        """
+        Tests tracker accuracy when using multiple parameter groups for
+        communication (for communication and computation overlap plus memory
+        reduction) and different mixed precision policies.
+        """
+        self.run_subtests(
+            {
+                "reshard_after_forward": [2],
+                "offload_policy": [
+                    CPUOffloadPolicy(pin_memory=False),
+                    OffloadPolicy(),
+                ],
+                "mp_policy": [
+                    MixedPrecisionPolicy(
+                        param_dtype=torch.float16, reduce_dtype=torch.float32
+                    ),
+                    MixedPrecisionPolicy(),
+                ],
+                "device_type": ["cuda"],
```

Comment:
nit: I think I included `device_type` before in case we cared about CPU training, but it turns out that we are pretty far from supporting that (a few different things we need to address). I guess you could exclude this arg and just directly use `"cuda"`.